sample/android/src/main/java/com/freeletics/flowredux/compose/PopularRepositoriesUi.kt

```diff
@@ -0,0 +1,62 @@
+package com.freeletics.flowredux.compose
+
+import androidx.compose.material.Scaffold
+import androidx.compose.material.SnackbarDuration
+import androidx.compose.material.rememberScaffoldState
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.ui.res.stringResource
+import com.freeletics.flowredux.AndroidFlowReduxLogger
+import com.freeletics.flowredux.R
+import com.freeletics.flowredux.sample.shared.*
+import kotlinx.coroutines.launch
+
+@Composable
+fun PopularRepositoriesUi() {
+    val stateMachine = InternalPaginationStateMachine(
+        githubApi = GithubApi(),
+        logger = AndroidFlowReduxLogger
+    )
+
+    val (state, dispatch) = stateMachine.stateAndDispatch()
+
+    // Timber.d("New State ${state.value}")
```

Comment:
@gabrielittner  here I run in an interesting issue. If I let this line execute, then on dispatching an action I get the following error message:


> java.lang.IllegalStateException: Cannot dispatch action com.freeletics.flowredux.sample.shared.RetryLoadingFirstPage@325bf06 because state Flow of this FlowReduxStateMachine is not collected yet. Start collecting the state Flow before dispatching any action.
>         at com.freeletics.flowredux.dsl.FlowReduxStateMachine.dispatch$suspendImpl(FlowReduxStateMachine.kt:67)
>         at com.freeletics.flowredux.dsl.FlowReduxStateMachine.dispatch(Unknown Source:0)
>         at com.freeletics.flowredux.compose.ComposeFlowReduxStateMachineKt$stateAndDispatch$1$1.invokeSuspend(ComposeFlowReduxStateMachine.kt:54)
>         at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
>         at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
>         at androidx.compose.ui.platform.AndroidUiDispatcher.performTrampolineDispatch(AndroidUiDispatcher.android.kt:81)
>         at androidx.compose.ui.platform.AndroidUiDispatcher.access$performTrampolineDispatch(AndroidUiDispatcher.android.kt:41)
>         at androidx.compose.ui.platform.AndroidUiDispatcher$dispatchCallback$1.run(AndroidUiDispatcher.android.kt:57)
>         at android.os.Handler.handleCallback(Handler.java:883)
>         at android.os.Handler.dispatchMessage(Handler.java:100)
>         at android.os.Looper.loop(Looper.java:214)
>         at android.app.ActivityThread.main(ActivityThread.java:7356)
>         at java.lang.reflect.Method.invoke(Native Method)
>         at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
>         at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930)

Obviously this is coming from one of our internal checks
https://github.com/freeletics/FlowRedux/blob/088ef8a4d5d78e81c015209c0df96cd9cc764914/dsl/src/commonMain/kotlin/com/freeletics/flowredux/dsl/FlowReduxStateMachine.kt#L65

But what it should mean is that the previously collected `Flow` is not collected anymore. That doesn't make any sense to me. Do you see how this is caused or where it is coming from?

Reply:
Nevermind, I figured it out. Just ignore my comment above.